docs/sdks/architecture/deployment-modes.md

```diff
@@ -1,152 +1,292 @@
+#### 1. Important targeting attributes are not available on the client
 
-Each company might weigh each of these considerations differently. The purpose of this page is to highlight common patterns across Eppo customers and discuss the tradeoffs on the dimensions above.
+If you have subject attributes that you cannot send to your client application, but is available on the server you can use [**offline initialization from precomputed assignments**](#offline-initialization-from-precomputed-assignments). In this approach, you'll pre-evaluate all flags on the server side and pass them down to the client as part of routine app initialization. All that is available on the client is a list of flag values (with their names obfuscated).
 
-## Common deployment patterns 
+#### 2. Individual flags have very complex rulesets
 
-### Local flag evaluation using configurations from CDN (recommended)
+In general, the rules file is very lightweight and can be loaded in a few milliseconds. That said, the file gets larger as the complexity of targeting increases. For human-created rules logic, this is typically not a concern. However, if you are using Eppo to run [Contextual Bandits](/contextual-bandits/) with a large amount of context and potential actions, this file can get large. 
 
-The simplest and least error prone approach is to load feature flag configurations from Eppo's CDN each time the SDK is initialized (with optional subsequent polling for updates). This only requires one network call to Eppo's CDN at app initialization, which typically returns in under 15ms. 
+To prevent loading this whole file to your application, you can instead evaluate the flag in an Edge function managed by Eppo. On SDK initialization, you'll include the relevant context and Eppo will determine which action to take remotely without ever sending the full rule-set to the client. We refer to this pattern as [**precomputed assignment from Edge functions**](#precomputed-assignments-from-edge-functions). 
 
-Eppo's SDK will then locally determine what variant a user should see based on this configuration. The logging callback function is invoked at the exact moment the user was exposed to the variant. This is ideal from a data analysis perspective as only users exposed to the variant will be included in the analysis. That is, there is as little dilution as possible.
+### Serverless architectures
+
+Finally, if you are using a serverless architecture, we recommend using a [**local evaluation from pre-fetched configurations**](#local-flag-evaluation-initialized-with-pre-fetched-configurations) pattern. In this pattern, you'll manage a cache of flag configurations and pass that as an argument when the SDK is initialized in the serverless function. By doing this, you avoid requesting the configuration from Eppo's CDN separately for every call to your function.
```

Comment:
```suggestion
Finally, if you are using a serverless architecture, we recommend using a [**local evaluation from pre-fetched configurations**](#local-flag-evaluation-initialized-with-pre-fetched-configurations) pattern. In this pattern, you'll manage a cache of flag configurations and pass the flag configurations as an argument when the SDK is initialized in the serverless function. By doing this, you avoid requesting the configuration from Eppo's CDN separately for every call to your function.
```

To help clarify that "that" doesn't refer to the cache itself, but the data it holds. The cache would be some other cloud storage service, e.g. https://aws.amazon.com/blogs/compute/caching-data-and-configuration-settings-with-aws-lambda-extensions/